components/modelregistry/modelregistry.go

```diff
@@ -0,0 +1,84 @@
+// Package modelregistry provides utility functions to config ModelRegistry, an ML Model metadata repository service
+package modelregistry
+
+import (
+	"context"
+	"path/filepath"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/components"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+var (
+	ComponentName = "model-registry-operator"
+	Path          = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/odh"
+)
+
+// Verifies that ModelRegistry implements ComponentInterface.
+var _ components.ComponentInterface = (*ModelRegistry)(nil)
+
+// ModelRegistry struct holds the configuration for the ModelRegistry component.
+// +kubebuilder:object:generate=true
+type ModelRegistry struct {
+	components.Component `json:""`
+}
+
+func (m *ModelRegistry) OverrideManifests(_ string) error {
+	// If devflags are set, update default manifests path
+	if len(m.DevFlags.Manifests) != 0 {
+		manifestConfig := m.DevFlags.Manifests[0]
+		if err := deploy.DownloadManifests(ComponentName, manifestConfig); err != nil {
+			return err
+		}
+		// If overlay is defined, update paths
+		defaultKustomizePath := "overlays/odh"
+		if manifestConfig.SourcePath != "" {
+			defaultKustomizePath = manifestConfig.SourcePath
+		}
+		Path = filepath.Join(deploy.DefaultManifestPath, ComponentName, defaultKustomizePath)
+	}
+
+	return nil
+}
+
+func (m *ModelRegistry) GetComponentName() string {
+	return ComponentName
+}
+
+func (m *ModelRegistry) ReconcileComponent(ctx context.Context, cli client.Client, resConf *rest.Config,
+	owner metav1.Object, dscispec *dsciv1.DSCInitializationSpec, _ bool) error {
+	var imageParamMap = map[string]string{
+		"IMAGES_MODELREGISTRY_OPERATOR": "RELATED_IMAGE_ODH_MODELREGISTRY_OPERATOR_IMAGE",
```

Comment:
need PR ready in midstream with these env variable before sync to RHOAI

Reply:
Thanks, can you point me to a similar PR for that? I'll open one for model-registry.

Reply:
i think the standard workflow is a jira to devops team to deal with any new component in RHOAI.
there was some docs about the process but i dont recall any detail.

Reply:
Thanks, I'll ping devops team about it.

Reply:
could not find this part has been done in CPaaS, otherwise, i am fine with the change in PR.

Reply:
to not let CPaaS become a blocker for merging to incubation, we can leave this part for later commits

Reply:
We are targeting the 2.8 release. We'll add these variables to that release in CPaaS.